From 896d61251048bed52d5b2eb6c3cdd6de25634b0b Mon Sep 17 00:00:00 2001
From: Miroslaw Swiader <miroslaw.swiader@intel.com>
Date: Tue, 9 Feb 2016 13:23:03 +0100
Subject: [PATCH 4/5] usb: dwc3: remote wakeup only for USB2.0

This patch is a workarround. According to Synopsys
spec it should be working for SS and USB2.0 but
for SS it doesn't.

Change-Id: I0f6b4723ef0146919d3ac96d98bff303a064cbd0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19753
Signed-off-by: Miroslaw Swiader <miroslaw.swiader@intel.com>
Reviewed-on: https://android.intel.com:443/468863
---
 drivers/usb/dwc3/gadget.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index b152c4b..1357bdc 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -1000,8 +1000,9 @@ static int __dwc3_gadget_kick_transfer(struct dwc3_ep *dep, u16 cmd_param,
 		params.param0 = upper_32_bits(req->trb_dma);
 		params.param1 = lower_32_bits(req->trb_dma);
 		cmd = DWC3_DEPCMD_STARTTRANSFER;
-		if ((dwc3_readl(dwc->regs, DWC3_DSTS)
-			& DWC3_DSTS_USBLNKST_MASK) != 0) {
+		if (dwc->gadget.speed != USB_SPEED_SUPER &&
+		  (dwc3_readl(dwc->regs, DWC3_DSTS)
+		  & DWC3_DSTS_USBLNKST_MASK) != 0) {
 			reg = dwc3_readl(dwc->regs, DWC3_DCTL);
 			reg &= ~(DWC3_DCTL_ULSTCHNGREQ_MASK);
 			dwc3_writel(dwc->regs, DWC3_DCTL, reg);
-- 
1.9.1

